saleor/plugins/webhook/tasks.py

```diff
@@ -60,37 +79,38 @@ def _get_webhooks_for_event(event_type, webhooks=None):
     return webhooks
 
 
-@app.task(compression="zlib")
-def trigger_webhooks_for_event(event_type, data):
-    """Send a webhook request for an event as an async task."""
+def trigger_webhooks_async(data, event_type):
+    payload = EventPayload.objects.create(payload=data)
     webhooks = _get_webhooks_for_event(event_type)
-    for webhook in webhooks:
-        send_webhook_request.delay(
-            webhook.app.name,
-            webhook.pk,
-            webhook.target_url,
-            webhook.secret_key,
-            event_type,
-            data,
-        )
+    deliveries = create_event_delivery_list_for_webhooks(
+        webhooks=webhooks,
+        event_payload=payload,
+        event_type=event_type,
+    )
+    for delivery in deliveries:
+        send_webhook_request.delay(delivery.id)
 
 
 def trigger_webhook_sync(event_type: str, data: str, app: "App"):
     """Send a synchronous webhook request."""
-    webhooks = _get_webhooks_for_event(event_type, app.webhooks.all())
-    webhook = webhooks.first()
-    if not webhook:
+    webhooks = _get_webhooks_for_event(event_type, app.webhooks.all())[:1]
```

Comment:
Question: Wouldn't be more readable to create EventDelivery here directly? I mean, from one side it is great that we have a single function for that with a bulk but from the other, we have to modify the input and output arguments to operate over a list - like list with single elements, plural name of the variables

Reply:
Yes I agree. fixed it